### PR TITLE
AG-12985 - Fix radial gauge download.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-12985

### CODE REVIEW

File: packages/ag-charts-community/src/module/optionsModule.ts [hunk 1]
```diff
@@ -266,7 +265,7 @@ export class ChartOptions<T extends AgChartOptions = AgChartOptions> {
         };
 
         const processedSeries = (options.series as SeriesOptionsTypes[])?.map((series) => {
-            series.type ??= defaultSeriesType;
+            series.type ??= this.getDefaultSeriesType(options);
```
Comment (alantreadway) [inline packages/ag-charts-community/src/module/optionsModule.ts:268]: NOTE: This now lazily calls `this.getDefaultSeriesType()` if there is no `series.type`.

File: packages/ag-charts-community/src/chart/mapping/types.ts [hunk 2]
```diff
@@ -42,19 +42,18 @@ export type SeriesOptionsTypes =
...
 export type SeriesType = SeriesOptionsTypes['type'];
 
-export function optionsType(input: { series?: { type?: SeriesType }[] }): NonNullable<SeriesType> {
-    return input.series?.[0]?.type ?? 'line';
+export function optionsType(input: { series?: { type?: SeriesType }[] }): SeriesType {
+    const { series } = input;
+    if (series) {
+        return input.series?.[0]?.type ?? 'line';
+    }
+    return undefined;
```
Comment (iMoses) [inline packages/ag-charts-community/src/chart/mapping/types.ts]:
```suggestion
    if (series) {
        return series?.[0]?.type ?? 'line';
    }
```
nit: cleaner + I'm on the side that thinks `undefined` shouldn't be used as a value 😄

**Review by iMoses — APPROVED** [review]
LGTM, one nit

File: packages/ag-charts-community/src/chart/mapping/types.ts (hunk 2 above)
Comment (lsjroberts) [reply packages/ag-charts-community/src/chart/mapping/types.ts]: My version:

```suggestion
    if (!series) return;
    return series[0]?.type ?? 'line';
```

**Review by jacobp100 — APPROVED** [review]

File: packages/ag-charts-community/src/chart/mapping/types.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/mapping/types.ts]: Thanks both, was meaning to clean this up.

**Pull request merged**